routingpy/routers/opentripplanner.py

```diff
@@ -0,0 +1,244 @@
+
+        :param retry_timeout: Timeout across multiple retriable requests, in seconds.
+            Default :attr:`routingpy.routers.options.default_retry_timeout`.
+        :type retry_timeout: int
+
+        :param retry_over_query_limit: If True, client will not raise an exception on HTTP 429,
+            but instead jitter a sleeping timer to pause between requests until HTTP 200 or
+            retry_timeout is reached.
+            Default :attr:`routingpy.routers.options.default_retry_over_query_limit`.
+        :type retry_over_query_limit: bool
+
+        :param skip_api_error: Continue with batch processing if a
+            :class:`routingpy.exceptions.RouterApiError` is encountered (e.g. no route found).
+            If False, processing will discontinue and raise an error.
+            Default :attr:`routingpy.routers.options.default_skip_api_error`.
+        :type skip_api_error: bool
+
+        :param client: A client class for request handling. Needs to be derived from
+            :class:`routingpy.client_base.BaseClient`
+        :type client: abc.ABCMeta
+
+        :param client_kwargs: Additional arguments passed to the client, such as headers or proxies.
+        :type client_kwargs: dict
+        """
+
+        self.client = client(
+            base_url,
+            user_agent,
+            timeout,
+            retry_timeout,
+            retry_over_query_limit,
+            skip_api_error,
+            **client_kwargs,
+        )
+
+    def directions(
+        self,
+        locations: List[List[float]],
+        profile: str,
+        alternatives: Optional[Union[bool, int]] = None,
+        dry_run: Optional[bool] = None,
+    ):
+        """
+        Get directions between an origin point and a destination point.
+
+        :param locations: The coordinates tuple the route should be calculated from in order of
+            visit.
+        :type locations: list of list
+
+        :param profile: Specifies the mode of transport to use when calculating directions. Possible
+            values are "CAR", "BICYCLE", "TRANSIT", "WALK". For more profiles, see OpenTripPlanner's
+            GraphiQL documentation.
+        :type profile: str
+
+        :param alternatives: Search for alternative routes. A result cannot be guaranteed. Accepts
+            an integer or False. Default False.
+        :type alternatives: bool
+
+        :param dry_run: Print URL and parameters without sending the request.
+        :type dry_run: bool
+
+        :returns: One or multiple route(s) from provided coordinates and restrictions.
+        :rtype: :class:`routingpy.direction.Direction` or :class:`routingpy.direction.Directions`
+        """
+        num_itineraries = 1 if not alternatives else 5
+        query = f"""
+            {{
+                plan(
+                    from: {{lat: {locations[0][1]}, lon: {locations[0][0]}}}
+                    to: {{lat: {locations[1][1]}, lon: {locations[1][0]}}}
```

Comment:
there is no intermediate waypoints implemented in OTP or we just don't expose them (yet)? fwiw, it's fine to not have them here in any case, someone else will do it if they need it

Reply:
I haven't checked because I didn't have the need yet. Maybe later